src/state_machine.zig

```diff
@@ -1929,73 +1911,178 @@ pub fn StateMachineType(
+                        .user_data_32 = .{ .batch_value_count_limit = batch_values_limit.accounts.user_data_32 },
+                        .ledger = .{ .batch_value_count_limit = batch_values_limit.accounts.ledger },
+                        .code = .{ .batch_value_count_limit = batch_values_limit.accounts.code },
                     },
                 },
                 .transfers = .{
                     // lookup_transfer() looks up 1 Transfer.
                     // create_transfer() looks up at most 1 Transfer for posting/voiding.
-                    .prefetch_entries_for_read_max = batch_transfers_max,
+                    .prefetch_entries_for_read_max = batch_transfers_limit,
                     // create_transfer() updates a single Transfer.
-                    .prefetch_entries_for_update_max = batch_transfers_max,
+                    .prefetch_entries_for_update_max = batch_transfers_limit,
                     .cache_entries_max = options.cache_entries_transfers,
-                    .tree_options_object = .{},
-                    .tree_options_id = .{},
+                    .tree_options_object = .{ .batch_value_count_limit = batch_values_limit.transfers.timestamp },
+                    .tree_options_id = .{ .batch_value_count_limit = batch_values_limit.transfers.id },
                     .tree_options_index = .{
-                        .debit_account_id = .{},
-                        .credit_account_id = .{},
-                        .user_data_128 = .{},
-                        .user_data_64 = .{},
-                        .user_data_32 = .{},
-                        .pending_id = .{},
-                        .ledger = .{},
-                        .code = .{},
-                        .amount = .{},
-                        .expires_at = .{},
+                        .debit_account_id = .{ .batch_value_count_limit = batch_values_limit.transfers.debit_account_id },
+                        .credit_account_id = .{ .batch_value_count_limit = batch_values_limit.transfers.credit_account_id },
+                        .user_data_128 = .{ .batch_value_count_limit = batch_values_limit.transfers.user_data_128 },
+                        .user_data_64 = .{ .batch_value_count_limit = batch_values_limit.transfers.user_data_64 },
+                        .user_data_32 = .{ .batch_value_count_limit = batch_values_limit.transfers.user_data_32 },
+                        .pending_id = .{ .batch_value_count_limit = batch_values_limit.transfers.pending_id },
+                        .ledger = .{ .batch_value_count_limit = batch_values_limit.transfers.ledger },
+                        .code = .{ .batch_value_count_limit = batch_values_limit.transfers.code },
+                        .amount = .{ .batch_value_count_limit = batch_values_limit.transfers.amount },
+                        .expires_at = .{ .batch_value_count_limit = batch_values_limit.transfers.expires_at },
```

Comment:
Feels like it's dangerously easy to make a typo here. Though I guess let's worry about that if we have to change the code